### PR TITLE
add new locale endpoint that returns the messages

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Unreported flaky test detected, please review

**Review by pedroigor — COMMENTED** [review]
Adding the `needs-discussion` label. As we discussed in private, it deserves a bit more discussion because of the public endpoint being introduced.

File: services/src/main/java/org/keycloak/services/resources/LocaleResource.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            return new ArrayList<>(result);
+        }
+        return theTheme.getEnhancedMessages(realm, locale).entrySet().stream().map(e ->
+                new KeySource((String) e.getKey(), (String) e.getValue())).collect(toList());
+    }
+}
+
+enum Source {
+    THEME,
+    OVERRIDE
```
Comment (ssilvert) [inline services/src/main/java/org/keycloak/services/resources/LocaleResource.java:65]: I think this should be REALM or maybe REALM_OVERRIDE.  The user needs to know where this was specified.  

Also, a realm key might exist where a theme key does not.  In that case, it's not really an override at all.

**Comment by ssilvert** [comment]
> Adding the `needs-discussion` label. As we discussed in private, it deserves a bit more discussion because of the public endpoint being introduced.

It's possible someone could put confidential info into a message bundle.  But I think it's unlikely and I don't think that concern would be worth changing this PR.  We could mitigate with a simple warning.  Users should not be putting confidential info into message bundles under any circumstances anyway.

The only other problem I see with a public endpoint for this is that maybe public endpoints are more prone to DOS attacks?  I don't know the answer to that or if it matters.

**Comment by pedroigor** [comment]
> > Adding the `needs-discussion` label. As we discussed in private, it deserves a bit more discussion because of the public endpoint being introduced.
> 
> It's possible someone could put confidential info into a message bundle. But I think it's unlikely and I don't think that concern would be worth changing this PR. We could mitigate with a simple warning. Users should not be putting confidential info into message bundles under any circumstances anyway.
> 
> The only other problem I see with a public endpoint for this is that maybe public endpoints are more prone to DOS attacks? I don't know the answer to that or if it matters.

Yeah, I'm more concerned about the latter as well as introducing a step when deploying in production to explain what it does, and how to protect in case you just want it accessible from your internal network.

I made a suggestion to @edewit to instead make this endpoint private in both Admin and Account APIs. That way we are a bit more careful before exposing a public-facing endpoint and everything around it.

**Comment by edewit** [comment]
We could change it for now to not be public, but in future we want to use this to also translate login, when we do that it needs to be public. FWI what we have atm is also public it's only "hidden" behind a cache key

**Comment by edewit** [comment]
@pedroigor made it non public, can it be merged like this

**Review by pedroigor — APPROVED** [review]

**Pull request merged**